core/templates/dev/head/collection_editor/CollectionEditor.js

```diff
@@ -25,32 +25,84 @@ oppia.constant(
 oppia.constant(
   'COLLECTION_RIGHTS_URL_TEMPLATE',
   '/collection_editor_handler/rights/<collection_id>');
+oppia.constant(
+  'EXPLORATION_SUMMARY_DATA_URL_TEMPLATE',
+  '/explorationsummarieshandler/data?' +
+  'stringified_exp_ids=<stringified_exp_ids>&' +
+  'include_private_explorations=<include_private_explorations>');
 
 oppia.controller('CollectionEditor', ['$scope',
   'WritableCollectionBackendApiService', 'CollectionRightsBackendApiService',
-  'CollectionObjectFactory', 'SkillListObjectFactory',
+  'ExplorationSummaryBackendApiService', 'CollectionObjectFactory',
+  'SkillListObjectFactory', 'CollectionValidationService',
   'CollectionUpdateService', 'UndoRedoService', 'alertsService', function(
     $scope, WritableCollectionBackendApiService,
-    CollectionRightsBackendApiService, CollectionObjectFactory,
-    SkillListObjectFactory, CollectionUpdateService, UndoRedoService,
+    CollectionRightsBackendApiService, ExplorationSummaryBackendApiService,
+    CollectionObjectFactory, SkillListObjectFactory,
+    CollectionValidationService, CollectionUpdateService, UndoRedoService,
     alertsService) {
     $scope.collection = null;
     $scope.collectionId = GLOBALS.collectionId;
     $scope.collectionSkillList = SkillListObjectFactory.create([]);
     $scope.isPublic = GLOBALS.isPublic;
+    $scope.validationIssues = [];
+
+    var _validateCollection = function() {
+      if ($scope.isPublic) {
```

Comment:
Good point. I switched over to sending whether the collection is private, instead of public. Thanks for pointing this out.